scripts/code_generation/jsoncan/src/codegen/c_generation/app_can_rx_module.py

```diff
@@ -1,23 +1,45 @@
 import jinja2 as j2
 
+from ...can_database import All, CanDatabase
+from .cmodule import CModule
 from .routing import CanRxConfig
-from ...can_database import CanDatabase
 from .utils import load_template
-from .cmodule import CModule
 
 
 class AppCanRxModule(CModule):
     def __init__(self, db: CanDatabase, node: str, rx_config: CanRxConfig):
         self._db = db
         self._node = node
-        self._rx_msgs = [self._db.msgs[msg_name] for msg_name in rx_config.get_all_rx_msgs_names()]
+        self._rx_msgs = [
+            self._db.msgs[msg_name] for msg_name in rx_config.get_all_rx_msgs_names()
+        ]
+        self._alert_boards_messages = {}
+        alerts_boards = db.alerts.keys()
+
+        for board in alerts_boards:
+            rx_msgs = []
+            if isinstance(db.nodes[node].rx_msgs_names, All): # if all then talk all the tx message for that board
+                rx_msgs = db.get_board_tx_messages(board)
+            else:
+                for msg in db.get_board_tx_messages(board): # else need to filter the message that is rx by the board
+                    if msg.name in db.nodes[node].rx_msgs_names:
+                        rx_msgs.append(msg)
+            self._alert_boards_messages[board] = rx_msgs
```

Comment:
why do we need to recalculate? this seems less trivial calculations and more structural reorganization